### PR TITLE
feat: Red alarm

### PR DESCRIPTION
**Pull request by ChefMomota**
- [x] Hide when is native token or chainId is not BSC
- [x] If create time more than 7 days. User need to fetch it again

### We need to wait until they update the latest response for us.
- [x] Change new response data

### CODE REVIEW

File: src/views/Swap/components/AccessRisk/index.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        [address]: {
+          ...data,
+          createDate: new Date().getTime(),
+        },
+      },
+    }
+
+    setRiskTokenInfo(data)
+    setIsFetchStatusSuccess(data.isSuccess)
+    localStorage.setItem(TOKEN_RISK_KEY, JSON.stringify(newData))
```
[the remark this answers is not in the document]
Comment (ChefMomota) [reply src/views/Swap/components/AccessRisk/index.tsx:81]: Actually is not all risk token. Just can give user check that token is risk or not.
So we will save the data, User will not need to keep checking if the token is risky.

File: src/views/Swap/components/AccessRisk/index.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  }, [currency])
+
+  const handleAccess = async () => {
+    setIsAccessing(true)
+
+    const { address, chainId, symbol } = currency as any
+    toastInfo(t('Accessing Risk'), t('Please wait until we scan the risk for %symbol% token', { symbol }))
+
+    const tokenRiskResult: RiskTokenInfo = await fetchRiskToken(address, chainId)
+    setIsAccessing(false)
```
[the remark this answers is not in the document]
Comment (ChefMomota) [reply src/views/Swap/components/AccessRisk/index.tsx:56]: nice catch

File: src/views/Swap/hooks/fetchTokenRisk.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  address: string
+  chainId: ChainId
+  isSuccess: boolean
+  riskLevel: TOKEN_RISK
+  riskResult: string
+  scannedTs: number
+}
+
+const fetchVerifyTokens = async (chainId: number) => {
+  const listUrl = VerifyTokensLink[chainId]
```
Comment (chef-jojo) [inline src/views/Swap/hooks/fetchTokenRisk.ts:34]: can just use `combinedTokenMapFromOfficialsUrlsSelector`

File: src/views/Swap/index.tsx [hunk 4]
```diff
@@ -385,6 +387,12 @@ export default function Swap() {
     [chainId],
   )
 
+  const isAccessTokenSupported = useMemo(() => ACCESS_TOKEN_SUPPORT_CHAIN_IDS.includes(chainId), [chainId])
+  const isShowAccessToken = useMemo(() => {
+    // TODO: Should verify with native token instead of BNB
+    return isAccessTokenSupported && currencies[Field.OUTPUT]?.symbol !== 'BNB'
```
Comment (chef-jojo) [inline src/views/Swap/index.tsx:393]: ?.isNative

Comment (ChefMomota) [reply src/views/Swap/index.tsx:393]: yup. will update

File: .env.development [hunk 5]
```diff
@@ -12,3 +12,5 @@ NEXT_PUBLIC_SNAPSHOT_BASE_URL = "https://testnet.snapshot.org"
 
 NEXT_PUBLIC_API_NFT = "https://nft.pancakeswap.com/api/v1"
 NEXT_PUBLIC_BIT_QUERY_ENDPOINT = "https://graphql.bitquery.io"
+
+NEXT_PUBLIC_ACCESS_RISK = "https://red.alert.pancakeswap.com/red-api"
```
Comment (chef-jojo) [inline .env.development:16]: No need to set in env

File: src/views/Swap/index.tsx [hunk 6]
```diff
@@ -411,6 +413,11 @@ export default function Swap() {
     [chainId],
   )
 
+  const isAccessTokenSupported = useMemo(() => ACCESS_TOKEN_SUPPORT_CHAIN_IDS.includes(chainId), [chainId])
+  const isShowAccessToken = useMemo(() => {
+    return isAccessTokenSupported && currencies[Field.OUTPUT]?.symbol !== NATIVE[chainId].symbol
```
Comment (chef-jojo) [inline src/views/Swap/index.tsx:418]: IsNative

**Review by chef-jojo — CHANGES_REQUESTED** [review]

File: src/views/Swap/index.tsx (hunk 6 above)
Comment (ChefMomota) [reply src/views/Swap/index.tsx]: @chef-jojo  done

File: .env.development (hunk 5 above)
Comment (ChefMomota) [reply .env.development]: done

File: src/views/Swap/components/AccessRisk/index.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    return false
+  }, [currency, riskTokenInfo, isScanning])
+
+  const handleScan = async () => {
+    setIsScanning(true)
+
+    const { address, chainId, symbol } = currency as any
+    toastInfo(t('Scanning Risk'), t('Please wait until we scan the risk for %symbol% token', { symbol }))
+
+    const tokenRiskResult: RiskTokenInfo = await fetchRiskToken(address, chainId)
```
Comment (chefilip) [inline src/views/Swap/components/AccessRisk/index.tsx:60]: should we use `useSWR` to cache the resquest

**Review by chefilip — APPROVED** [review]

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**